### PR TITLE
Websocket support

### PR DESCRIPTION
**Pull request by jukkar**
The websocket support was originally introduced by PR #1271 but that PR was closed because it was too convoluted. Introducing the websocket support again. This PR contains relevant websocket related changes from PR #5035

### CODE REVIEW

File: drivers/console/CMakeLists.txt [hunk 1]
```diff
@@ -7,4 +7,4 @@ zephyr_sources_if_kconfig(uart_mcumgr.c)
 zephyr_sources_if_kconfig(uart_pipe.c)
 zephyr_sources_if_kconfig(telnet_console.c)
 zephyr_sources_if_kconfig(xtensa_sim_console.c)
-zephyr_sources_if_kconfig(native_posix_console.c)
```
Comment (nashif) [inline drivers/console/CMakeLists.txt]: why are you removing the native_posix console?

**Review by nashif — DISMISSED** [review]

File: drivers/console/CMakeLists.txt (hunk 1 above)
Comment (jukkar) [reply drivers/console/CMakeLists.txt]: Some error while rebasing, @rveerama1 will send a fixed version shortly

Comment (rveerama1) [reply drivers/console/CMakeLists.txt]: I have been maintaining websocket patches in RPL BR sample. It was my mistake in rebasing things. I will fix it. Thanks @nashif

Comment (rveerama1) [reply drivers/console/CMakeLists.txt]: Fixed and pushed updated version.

**Comment by rveerama1** [comment]
Fixed CI errors.

File: samples/net/ws_console/README.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/ws_console
+   :board: qemu_x86
+   :goals: run
+   :compact:
+
+The default make BOARD configuration for this sample is ``qemu_x86``.
+
+Connect to the console from your browser using these URLs http://[2001:db8::1]
+or http://192.0.2.1
```
Comment (dbkinder) [inline samples/net/ws_console/README.rst]: at the end of the sentence, add:

    (as configured in the project's ``prj.conf`` file).

File: samples/net/ws_echo_server/README.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+Websocket Server
+################
+
+Overview
+********
+
+The websocket-server sample application for Zephyr implements a websocket
+server. The websocket-server listens for incoming IPv4 or IPv6 HTTP(S)
+requests.
```
Comment (dbkinder) [inline samples/net/ws_echo_server/README.rst]: Add what it does in response to the incoming requests.
It also appears you need to use a web browser to interact with this sample, but you don't talk about that here.

File: samples/net/ws_echo_server/README.rst [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Build websocket-server sample application like this:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/ws_echo_server
+   :board: qemu_x86
+   :goals: run
+   :compact:
+
+The default make BOARD configuration for this sample is ``qemu_x86``.
```
Comment (dbkinder) [inline samples/net/ws_echo_server/README.rst]: It appears you interact with this sample app with a web browser, but you don't include that in these instructions.

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by rveerama1** [comment]
@dbkinder addressed your comments.

**Review by dbkinder — APPROVED** [review]
Thanks!

**Review by rveerama1 — APPROVED** [review]

**Pull request merged**